### PR TITLE
[1LP][RFR] New Test: Testing expression method with dialog

### PR DESCRIPTION
**Pull request by ganeshhubale**
Signed-off-by: Ganesh Hubale <ghubale@redhat.com>

## Purpose or Intent
- Testing expression method for collecting vms associated with rhvem provider using dialog
### PRT Run
{{ pytest: cfme/tests/services/test_dialog_element_in_catalog.py::test_service_dialog_expression_method --long-running --use-template-cache -qsvvv --use-provider=complete }}

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 81d196b65908f41197a0f358ce16b348aecd4618

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/services/test_dialog_element_in_catalog.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/services/test_dialog_element_in_catalog.py [hunk 1]
```diff
@@ -820,7 +841,21 @@ def test_service_dialog_expression_method():
...
+        display_in=True,
+        catalog=catalog,
+        dialog=sd,
+    )
+    request.addfinalizer(catalog_item.delete_if_exists)
+    service_catalogs = ServiceCatalogs(
+        appliance, catalog=catalog_item.catalog, name=catalog_item.name
+    )
+    view = navigate_to(service_catalogs, "Order")
+    assert full_vm.name in [opt.text for opt in view.fields("dropdown_list_1").dropdown.all_options]
```
Comment (digitronik) [inline cfme/tests/services/test_dialog_element_in_catalog.py]: Q: is it ok to use `has_item` of `Dropdown` widget to check vm name?

File: cfme/tests/services/test_dialog_element_in_catalog.py [hunk 2]
```diff
@@ -793,12 +796,30 @@ def test_service_service_vms_retires_archived():
     pass
 
 
-@pytest.mark.meta(coverage=[1558926])
-@pytest.mark.manual
+@pytest.fixture(scope='function')
+def full_vm(appliance, provider, full_template):
```
Comment (sbulage) [inline cfme/tests/services/test_dialog_element_in_catalog.py]: Question:
Why do create/write `full_vm` function scoped fixture, as we already have one .?

File: cfme/tests/services/test_dialog_element_in_catalog.py [hunk 3]
```diff
@@ -820,7 +841,21 @@ def test_service_dialog_expression_method():
             4.
             5. Expression method should work
     """
-    pass
+    sd, ele_label = import_dialog
+    catalog_item = appliance.collections.catalog_items.create(
```
Comment (sbulage) [inline cfme/tests/services/test_dialog_element_in_catalog.py]: Suggestion:
`Catalog_item` should have finalizer.

Comment (sbulage) [reply cfme/tests/services/test_dialog_element_in_catalog.py]: My Bad. It's already there. Never mind

File: cfme/tests/services/test_dialog_element_in_catalog.py (hunk 1 above)
Comment (ganeshhubale) [reply cfme/tests/services/test_dialog_element_in_catalog.py]: @digitronik 
This dropdown is of widget type ```BootstrapSelect```. Hence I am getting below error:
```ipdb> view.fields("dropdown_list_1").dropdown.has_item(full_template_vm.name)                                                                                                
*** AttributeError: 'BootstrapSelect' object has no attribute 'has_item'
```

File: cfme/tests/services/test_dialog_element_in_catalog.py (hunk 2 above)
Comment (ganeshhubale) [reply cfme/tests/services/test_dialog_element_in_catalog.py]: Thanks for this suggestion.

**Pull request merged**